### PR TITLE
Basic login

### PR DESCRIPTION
**Pull request by kalpeshDocFox**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,4 +1,24 @@
 class UsersController < ApplicationController
+  def show
+    @user = User.find(params[:id])
+  end
+
   def new
+    @user = User.new
+  end
+  def create
```
Comment (c13e) [inline app/controllers/users_controller.rb]:
```suggestion
  end

  def create
```
please add new lines between methods, ditto for rest of this file

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module SessionsHelper
+  # Returns the current logged-in user (if any).
+  def current_user
+    if session[:user_id]
+      @current_user ||= User.find_by(id: session[:user_id])
+    end
+  end
+  # Logs in the given user.
+  def log_in(user)
```
Comment (c13e) [inline app/helpers/sessions_helper.rb]:
```suggestion
  end

  # Logs in the given user.
  def log_in(user)
```
ditto for new line between methods. Please go through rest of this PR and apply the same

File: public/packs-test/manifest.json [hunk 3]
```diff
@@ -1,13 +1,13 @@
 {
-  "application.js": "/packs-test/js/application-c28cbe491b13cb99de76.js",
-  "application.js.map": "/packs-test/js/application-c28cbe491b13cb99de76.js.map",
+  "application.js": "/packs-test/js/application-b1ba4b11862793390146.js",
```
Comment (c13e) [inline public/packs-test/manifest.json:2]: please add `public/packs-test` to your `.gitignore` file, these should not be committed

File: test/integration/users_login_test.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
+require 'test_helper'
+
+class UsersLoginTest < ActionDispatch::IntegrationTest
+  def setup
+    @user = users(:michael)
+  end
+
+  test "POST login with invalid login credentials" do
```
Comment (c13e) [inline test/integration/users_login_test.rb]:
```suggestion
  test "POST login with invalid login credentials should ..." do
```
add what the main assertions are to test name, ditto for other tests in this file

File: test/integration/users_login_test.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    assert_not flash.empty?
+    get root_path
+    assert flash.empty?
+  end
+
+  test "POST login with valid credentials" do
+    post login_path, params: { session: { email: @user.email, password: 'password' } }
+    assert_redirected_to @user
+  end
+  test "login with valid information followed by logout" do
```
Comment (c13e) [inline test/integration/users_login_test.rb]:
```suggestion
  end

  test "login with valid information followed by logout" do
```

File: test/integration/users_signup_test.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    assert_template 'users/new'
+  end
+
+  test "test to confirm with valid signup information" do
+    get signup_path
+    assert_difference 'User.count', 1 do
+      post users_path, params: { user: { name: "Example User",
+                                         email: "user@example.com",
+                                         password:"password",
+                                         confirm_password:"password" }}
```
Comment (c13e) [inline test/integration/users_signup_test.rb]:
```suggestion
      post users_path, params: params
```
extract params into a well formatted variable & use that, ex:
```rb
params = {
  user: {
    ...
  }
}
```

**Review by c13e — CHANGES_REQUESTED** [review]
Nice work, see my comments re: gitignoring some files that should not be committed

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (kalpeshDocFox) [reply app/controllers/users_controller.rb]: Noted added to all controllers and tests

Comment (kalpeshDocFox) [reply app/controllers/users_controller.rb]: added spaces

**Pull request merged**